compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1387,6 +1387,7 @@ public void visit(BLangSimpleVariable varNode) {
     @Override
     public void visit(BLangLetExpression letExpression) {
         SymbolEnv prevEnv = this.env;
+        letExpression.env.enclInvokable = this.env.enclInvokable;
```

Comment:
Why set this in Desugar? 🤔  Shouldn't this already be automatically set when the let expr env was created? 🤔

Reply:
It is in local level. But when in module level env.enclInvokable it sets to null

Reply:
Resolving as per the offline discussion.